go/ct/spc/specification.go

```diff
@@ -126,7 +130,6 @@ func getAllRules() []Rule {
 		),
 		Effect: Change(func(s *st.State) {
 			s.Status = st.Stopped
-			s.ReturnData = Bytes{}
 			s.Pc++
```

Comment:
This is surprising - when running into a "STOP" instruction -- explicit or implicit -- the return data should be empty.

I think what is happening is this: 
- in all implementations the (implicit) return data slice is always empty until a `RETURN` or a `REVERT` is reached. Only those two instructions are setting the return data to something that is not empty.
- in all implementations, processing a `STOP` thus does not need to explicitly set the return data to the empty data, since it is empty throughout the contract execution
- when we create a test state with the CT we will the return data with random values; those are now preserved by all implementations when running the `STOP` instruction, and it looks like `STOP` is not supposed to have an effect on the `ReturnData` -- while it should, by making sure the return data is empty; codes terminated by an explicit or implicit `STOP` must return no return data

So it looks like the root of the problem is actually the fact that we produce states with random return data. This change is "covering" this up, by -- what looks like -- an invalid specification. Instead I think the state generation should be updated to only produce empty return data. Wdyt?

Reply:
from https://www.evm.codes/:
```
STOP: When a call is executed on an address with no code and the EVM tries to read the code data, the default value is returned, 0, which corresponds to this instruction and halts the execution.
```
I think this is exactly what Simon describes: a call into a no-code address. 
can we interpret here that 0 needs to be returned?

Reply:
I also think the return data should not be set in the state generation as it is never read and only set by RETURN/REVERT opcodes.
The random return data is moved from the state generation and the regression input.

Reply:
@LuisPH3  just pointed out that this makes sense for the RETURN/REVERT instructions but does not make sense for the RETURNDATACOPY/RETURNDATASIZE instructions. A possible solution would be a special case in the state generation but I don't think this is the most elegant solution.